jwtmiddleware.go

```diff
@@ -84,38 +103,116 @@ func (m *JWTMiddleware) Handler(h http.Handler) http.Handler {
 	})
 }
 
-func (m *JWTMiddleware) CheckJWT(w http.ResponseWriter, r *http.Request) error {
-
+// FromAuthHeader is a "TokenExtractor" that takes a give request and extracts
+// the JWT token from the Authorization header.
+func FromAuthHeader(r *http.Request) (string, error) {
 	authHeader := r.Header.Get("Authorization")
 	if authHeader == "" {
-		if m.Options.CredentialsOptional {
-			return nil
-		}
-		errorMsg := "Authorization header isn't sent"
-		m.Options.ErrorHandler(w, r, errorMsg)
-		return errors.New(errorMsg)
+		return "", nil // No error, just no token
 	}
 
+	// TODO: Make this a bit more robust, parsing-wise
 	authHeaderParts := strings.Split(authHeader, " ")
 	if len(authHeaderParts) != 2 || strings.ToLower(authHeaderParts[0]) != "bearer" {
-		errorMsg := "Authorization header format must be Bearer {token}"
-		m.Options.ErrorHandler(w, r, errorMsg)
-		return errors.New(errorMsg)
+		return "", fmt.Errorf("Authorization header format must be Bearer {token}")
```

Comment:
It is returning an error.  `fmt.Errorf` produces an error object.  It does not output to console.  You are thinking of `fmt.Printf`.  `fmt.Errorf` is just a convenience function that does a `fmt.Sprintf` and then wraps the resulting string in an error object (e.g., `errors.New(...)`).